### PR TITLE
set default user variable to the user cmd runs as

### PR DESCRIPTION
**Pull request by gtmanfred**
### What issues does this PR fix or reference?
#39360

### Tests written?

No

### CODE REVIEW

File: salt/config/__init__.py [hunk 1]
```diff
@@ -1184,7 +1185,7 @@ def _gather_buffer_space():
     'zmq_backlog': 1000,
     'pub_hwm': 1000,
     'auth_mode': 1,
-    'user': 'root',
+    'user': getpass.getuser(),
```
Comment (cachedout) [inline salt/config/__init__.py]: I can't find any clear indication of whether or not this is fully supported on Windows. Could you please double-check?

Comment (gtmanfred) [inline salt/config/__init__.py]: Yup, i will do it first thing tomorrow.

**Comment by bbinet** [comment]
Thanks @gtmanfred, this looks good to me.

**Comment by bbinet** [comment]
Could it also be backported to at least 2016.11 branch so that it could be fixed in 2011.6.3?

**Comment by gtmanfred** [comment]
since you can set 'user' in /etc/salt/{minion,master} i tend to think that we should leave it as is in older versions.  but i am going to leave it up to Mike.

**Comment by cachedout** [comment]
I am OK with backporting this. I will mark it as such. Thanks.

**Comment by rallytime** [comment]
Done in #39429

**Pull request merged**